lib/Models/ArcGisFeatureServerCatalogItem.ts

```diff
@@ -0,0 +1,621 @@
+  }
+
+  @computed get name() {
+    if (this._featureServer.name && this._featureServer.name.length > 0) {
+      return this._featureServer.name;
+    }
+  }
+
+  @computed get dataCustodian() {
+    if (
+      this._featureServer.documentInfo &&
+      this._featureServer.documentInfo.author &&
+      this._featureServer.documentInfo.author.length > 0
+    ) {
+      return this._featureServer.documentInfo.author;
+    }
+  }
+
+  @computed get rectangle() {
+    const extent = this._featureServer.extent;
+
+    if (
+      isDefined(extent) &&
+      extent.spatialReference &&
+      extent.spatialReference.wkid
+    ) {
+      const wkid = "EPSG:" + extent.spatialReference.wkid;
+      if (!isDefined((proj4definitions as any)[wkid])) {
+        return undefined;
+      }
+
+      const source = new proj4.Proj((proj4definitions as any)[wkid]);
+      const dest = new proj4.Proj("EPSG:4326");
+
+      let p = proj4(source, dest, [extent.xmin, extent.ymin]);
+
+      const west = p[0];
+      const south = p[1];
+
+      p = proj4(source, dest, [extent.xmax, extent.ymax]);
+
+      const east = p[0];
+      const north = p[1];
+
+      const rectangle = { west: west, south: south, east: east, north: north };
+      return createStratumInstance(RectangleTraits, rectangle);
+    }
+
+    return undefined;
+  }
+
+  @computed get info() {
+    function newInfo(name: string, content?: string) {
+      const traits = createStratumInstance(InfoSectionTraits);
+      runInAction(() => {
+        traits.name = name;
+        traits.content = content;
+      });
+      return traits;
+    }
+
+    return [
+      newInfo(
+        i18next.t("models.arcGisMapServerCatalogItem.dataDescription"),
+        this._featureServer.description
+      ),
+      newInfo(
+        i18next.t("models.arcGisMapServerCatalogItem.copyrightText"),
+        this._featureServer.copyrightText
+      )
+    ];
+  }
+
+  @computed get legends(): StratumFromTraits<LegendTraits>[] | undefined {
+    if (
+      !this._item.useStyleInformationFromService ||
+      !this._featureServer.drawingInfo
+    ) {
+      return undefined;
+    }
+    const renderer = this._featureServer.drawingInfo.renderer;
+    const rendererType = renderer.type;
+
+    if (rendererType === "classBreaks") {
+      const classBreaksRenderer = <ClassBreaksRenderer>(
+        this._featureServer.drawingInfo.renderer
+      );
+
+      const items = [];
+      for (var i = 0; i < classBreaksRenderer.classBreakInfos.length; i++) {
+        const label = classBreaksRenderer.classBreakInfos[i].label;
+        const color = classBreaksRenderer.classBreakInfos[i].symbol.color;
+        if (color) {
+          const item = {
+            title: label,
+            color: convertEsriColorToCesiumColor(color).toCssColorString()
+          };
+          items.push(item);
+        }
+      }
+      const legend = <StratumFromTraits<LegendTraits>>(
+        (<unknown>{ items: items })
```

Comment:
Could possibly add the legend title here from `drawingInfo.renderer.field` - I think this is how Esri does it

Reply:
I agree that a legend title would be good... not too sure how to actually add one though so I might leave this for another time.